### PR TITLE
Getter for initial user of messaging

### PR DESCRIPTION
**Pull request by breathx**
By @LouiseMedova's request:

Provides an ability to get the id of original user who initiated communication with blockchain, during which, currently processing message was created.

Integration for `gear-test` won't be realized, until we hadn't done with #649 

@gear-tech/dev

### CODE REVIEW

File: pallets/gear/src/lib.rs [hunk 1]
```diff
@@ -453,9 +457,8 @@ pub mod pallet {
             while let Some(dispatch) = common::dequeue_dispatch() {
                 // Update message gas limit for it may have changed in the meantime
 
-                let gas_limit = T::GasHandler::get_limit(*dispatch.message_id())
-                    .map(|(gas, _id)| gas)
-                    .unwrap_or(0);
+                let (gas_limit, initiator) = T::GasHandler::get_limit(*dispatch.message_id())
```
Comment (SabaunT) [inline pallets/gear/src/lib.rs:460]: :thumbsup:

**Review by SabaunT — APPROVED** [review]

**Comment by SabaunT** [comment]
maybe use `origin`instead of `initiator`?

File: core-processor/src/processor.rs [hunk 2]
```diff
@@ -37,12 +40,19 @@ pub fn process<A: ProcessorExt + EnvExt + Into<ExtInfo> + 'static, E: Environmen
     dispatch: Dispatch,
     block_info: BlockInfo,
     existential_deposit: u128,
+    initiator: ProgramId,
 ) -> Vec<JournalNote> {
     if let Some(actor) = actor {
         let execution_settings = ExecutionSettings::new(block_info, existential_deposit);
+        let execution_context = ExecutionContext { initiator };
```
Comment (shamilsan) [inline core-processor/src/processor.rs:47]: Can we combine settings and context into one entity?

File: core-processor/src/processor.rs [hunk 3]
```diff
@@ -67,16 +77,28 @@ pub fn process_many<A: ProcessorExt + EnvExt + Into<ExtInfo> + 'static, E: Envir
...
     existential_deposit: u128,
+    // Will go away some time soon
+    initiators: Vec<ProgramId>,
 ) -> Vec<JournalNote> {
     let mut journal = Vec::new();
 
+    assert_eq!(dispatches.len(), initiators.len());
+    let mut initiators = initiators.into_iter();
+
     for dispatch in dispatches {
```
Comment (shamilsan) [inline core-processor/src/processor.rs:88]: Isn't better to use `zip` here?

File: core-processor/src/processor.rs (hunk 2 above)
Comment (breathx) [reply core-processor/src/processor.rs:47]: I don't think so.
Setting generate at the start of the queue processing and don't change from message to message inside the same block, while context relates to each message separately and will contain more data in future, e.g. available gas

Comment (shamilsan) [reply core-processor/src/processor.rs:47]: Got it, ok.

File: core-processor/src/processor.rs (hunk 3 above)
Comment (breathx) [reply core-processor/src/processor.rs]: Sure! Done

**Comment by breathx** [comment]
Still considering about naming:
* `gr_initiator`
* `gr_origin`
* `gr_origin_user`

What do you think, @LouiseMedova, which one is the best?

**Comment by LouiseMedova** [comment]
> Still considering about naming:
> 
> * `gr_initiator`
> * `gr_origin`
> * `gr_origin_user`
> 
> What do you think, @LouiseMedova, which one is the best?



> Still considering about naming:
> 
> * `gr_initiator`
> * `gr_origin`
> * `gr_origin_user`
> 
> What do you think, @LouiseMedova, which one is the best?

I think `gr_origin` is the best one.

**Pull request merged**